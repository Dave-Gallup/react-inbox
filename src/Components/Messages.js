import React, { Component } from 'react';
import Message from './Message.js'

class Messages extends Component{


  generateMessages(){

    var messages = [];

    for(let key in this.props.messages){

      messages.push(
        <Message
        updateState={this.props.updateState}
        subject={this.props.messages[key].subject} labels={this.props.messages[key].labels} starred={this.props.messages[key].starred} read={this.props.messages[key].read} selected={this.props.messages[key].selected} id={this.props.messages[key].id}
        key={this.props.messages[key].id}/>
      );
    }
    return messages;
  }

  render(){
    return (
      <div className="messages-list">
        {this.generateMessages()}
      </div>
    );
  }
}


export default Messages;