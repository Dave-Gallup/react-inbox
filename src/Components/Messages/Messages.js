import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Message from '../Message/Message.js';
import { fetchMessages } from '../../actions/actions';
import { Route } from 'react-router-dom';

export class Messages extends Component{

  componentDidMount() {
    this.props.fetchMessages();
  }

  render(){
    return (
      <div className="messages-list">
      {this.props.messageList.map(id => <Route path='/' key={id} render={(props) => (
        <Message {...props} key={id} id={id} />
      )}/>)}
      </div>
    );
  }
}

export const mapStateToProps = state => {
  const messageMap = state.messages.messageMap;
  const messageList = state.messages.messageList;
  return {
    messageMap,
    messageList
  }
};

export const mapDispatchToProps = dispatch => bindActionCreators({
  fetchMessages
},
dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Messages);

