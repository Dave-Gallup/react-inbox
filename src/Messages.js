import React, { Component } from 'react';
import Message from './Message.js'

class Messages extends Component{

  render(){
    return (
      <div className="messages-list">
        {this.props.messages.map((el) => <Message subject={el.subject} labels={el.labels} starred={el.starred} read={el.read} selected={el.selected} key={el.id}/>)}
      </div>
    )
  }
}


export default Messages;