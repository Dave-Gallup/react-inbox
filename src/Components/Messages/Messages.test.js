import React from 'react';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Messages, mapStateToProps } from './Messages';

describe('Messages', () => {
  it('should render without any props other than empty messageList defined', () => {
    const component = shallow(
      <Messages messageList={[]}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  })

  it('should render with messageList defined and populated', () => {
    const component = shallow(
      <Messages messageList={[1,2,3]}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  })



});