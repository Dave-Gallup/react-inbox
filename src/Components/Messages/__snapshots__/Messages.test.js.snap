// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages should render with messageList defined and populated 1`] = `
<div
  className="messages-list"
>
  <Route
    path="/"
    render={[Function]}
  />
  <Route
    path="/"
    render={[Function]}
  />
  <Route
    path="/"
    render={[Function]}
  />
</div>
`;

exports[`Messages should render without any props other than empty messageList defined 1`] = `
<div
  className="messages-list"
/>
`;
